### PR TITLE
Add timeout config for yarn connection health indicator

### LINKED ISSUE #470: Add timeout for YarnConnectionHealthIndicator
If the Yarn Url is unreachable, the /admin/health endpoint will suffer from a timeout. A configurable timeout should be added for the connection. `URLConnection.setConnectTimeout` can be used to implement it

**New application property**
```
health.yarnConnection.timeoutMillis
```
Default value = 60000

### PR DESCRIPTION
**Pull request by kevinwallimann**
Closes #470 

**New application property**
```
health.yarnConnection.timeoutMillis
```
Default value = 60000

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -32,6 +32,7 @@ javamelody.management-endpoint-monitoring-enabled=true
 # Health check settings
 health.databaseConnection.timeoutMillis=120000
 health.yarnConnection.testEndpoint=/cluster/cluster
+health.yarnConnection.timeoutMillis=120000
```
Comment (jozefbakus) [inline src/main/resources/application.properties]: Could you please add mention of this prop to the PR description? 
When I do release notes it is easier to scrap changes

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -32,6 +32,7 @@ javamelody.management-endpoint-monitoring-enabled=true
 # Health check settings
 health.databaseConnection.timeoutMillis=120000
```
Comment (jozefbakus) [inline src/main/resources/application.properties]: I would decrease timeouts to a max of 1 minute.

File: src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/utilities/Configs.scala [hunk 3]
```diff
@@ -122,8 +122,11 @@ object JobDefinitionConfig {
 object HealthConfig {
   lazy val databaseConnectionTimeoutMillis: Int =
     Try(Configs.conf.getInt("health.databaseConnection.timeoutMillis")).getOrElse(120000)
-  lazy val yarnConnectionTestEndpoint: String =
+  lazy val yarnConnectionTestEndpoint: String = {
     Configs.conf.getString("health.yarnConnection.testEndpoint")
+  }
+  lazy val yarnConnectionTimeoutMillisOpt: Option[Int] =
```
Comment (jozefbakus) [inline src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/utilities/Configs.scala:128]: What about no option with default value like in databaseConnectionTimeoutMillis

Comment (kevinwallimann) [reply src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/utilities/Configs.scala:128]: You don't have to set a connection timeout on the `URLConnection`. If I provide a default value, you cannot configure that behaviour anymore.

File: src/main/resources/application.properties (hunk 1 above)
Comment (kevinwallimann) [reply src/main/resources/application.properties]: Done

File: src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/utilities/Configs.scala (hunk 3 above)
Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/utilities/Configs.scala:128]: But why the different behavior for yarn compared to DB?

File: src/main/resources/application.properties (hunk 2 above)
Comment (kevinwallimann) [reply src/main/resources/application.properties]: Done

**Review by jozefbakus — APPROVED** [review]

**Pull request merged**